src/main/java/com/marianbastiurea/lifeofbees/Hive.java

```diff
@@ -351,12 +297,12 @@ public void addNewEggsFrameInHive(int year) {
 
 
         int eggsFrameFull = 0;
-        for (int i = 0; i < eggsFrames.size(); i++) {
-            if (this.eggsFrames.get(i).getNumberOfEggs() == maxEggPerFrame && eggsFrameFull < eggsFrames.size()) {
+        for (int i = 0; i < eggFrames.size(); i++) {
+            if (this.eggFrames.get(i).getNumberOfEggs() == this.eggFrames.get(i).getMaxEggPerFrame() && eggsFrameFull < eggFrames.size()) {
```

Comment:
Move logic `this.eggFrames.get(i).getNumberOfEggs() == this.eggFrames.get(i).getMaxEggPerFrame()` in new method named `isEggFrameFull(){ return getNumberOfEggs() == getMaxEggPerFrame()` in EggFrame class